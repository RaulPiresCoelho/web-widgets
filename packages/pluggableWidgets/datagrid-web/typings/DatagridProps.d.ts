/**
 * This file was generated from Datagrid.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Framework Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
<<<<<<< HEAD
import { ActionValue, DynamicValue, EditableValue, ListValue, ListActionValue, ListAttributeValue, ListAttributeListValue, ListExpressionValue, ListReferenceValue, ListReferenceSetValue, ListWidgetValue, SelectionSingleValue, SelectionMultiValue } from "mendix";
=======
import { ActionValue, DynamicValue, EditableValue, ListValue, ListActionValue, ListAttributeValue, ListExpressionValue, ListReferenceValue, ListReferenceSetValue, ListWidgetValue, SelectionSingleValue, SelectionMultiValue } from "mendix";
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
import { Big } from "big.js";

export type ItemSelectionMethodEnum = "checkbox" | "rowClick";

export type ItemSelectionModeEnum = "toggle" | "clear";

<<<<<<< HEAD
export type LoadingTypeEnum = "spinner" | "skeleton";

=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
export type ShowContentAsEnum = "attribute" | "dynamicText" | "customContent";

export type HidableEnum = "yes" | "hidden" | "no";

export type WidthEnum = "autoFill" | "autoFit" | "manual";

export type MinWidthEnum = "auto" | "minContent" | "manual";

export type AlignmentEnum = "left" | "center" | "right";

export interface ColumnsType {
    showContentAs: ShowContentAsEnum;
<<<<<<< HEAD
    attribute?: ListAttributeValue<string | Big | boolean | Date> | ListAttributeListValue<string | Big | boolean | Date>;
=======
    attribute?: ListAttributeValue<string | Big | boolean | Date>;
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    content?: ListWidgetValue;
    dynamicText?: ListExpressionValue<string>;
    exportValue?: ListExpressionValue<string>;
    header?: DynamicValue<string>;
    tooltip?: ListExpressionValue<string>;
    filter?: ReactNode;
    filterAssociation?: ListReferenceValue | ListReferenceSetValue;
    filterAssociationOptions?: ListValue;
<<<<<<< HEAD
    fetchOptionsLazy: boolean;
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    filterAssociationOptionLabel?: ListExpressionValue<string>;
    visible: DynamicValue<boolean>;
    sortable: boolean;
    resizable: boolean;
    draggable: boolean;
    hidable: HidableEnum;
    allowEventPropagation: boolean;
    width: WidthEnum;
    minWidth: MinWidthEnum;
    minWidthLimit: number;
    size: number;
    alignment: AlignmentEnum;
    columnClass?: ListExpressionValue<string>;
    wrapText: boolean;
}

export type PaginationEnum = "buttons" | "virtualScrolling" | "loadMore";

<<<<<<< HEAD
export type ShowPagingButtonsEnum = "always" | "auto";

export type PagingPositionEnum = "bottom" | "top" | "both";

=======
export type PagingPositionEnum = "bottom" | "top" | "both";

export type ShowPagingButtonsEnum = "always" | "auto";

>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
export type ShowEmptyPlaceholderEnum = "none" | "custom";

export type OnClickTriggerEnum = "single" | "double";

export type ConfigurationStorageTypeEnum = "attribute" | "localStorage";

export interface FilterListType {
    filter: ListAttributeValue<string | Big | boolean | Date>;
}

export interface ColumnsPreviewType {
    showContentAs: ShowContentAsEnum;
    attribute: string;
    content: { widgetCount: number; renderer: ComponentType<{ children: ReactNode; caption?: string }> };
    dynamicText: string;
    exportValue: string;
    header: string;
    tooltip: string;
    filter: { widgetCount: number; renderer: ComponentType<{ children: ReactNode; caption?: string }> };
    filterAssociation: string;
    filterAssociationOptions: {} | { caption: string } | { type: string } | null;
<<<<<<< HEAD
    fetchOptionsLazy: boolean;
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    filterAssociationOptionLabel: string;
    visible: string;
    sortable: boolean;
    resizable: boolean;
    draggable: boolean;
    hidable: HidableEnum;
    allowEventPropagation: boolean;
    width: WidthEnum;
    minWidth: MinWidthEnum;
    minWidthLimit: number | null;
    size: number | null;
    alignment: AlignmentEnum;
    columnClass: string;
    wrapText: boolean;
}

export interface FilterListPreviewType {
    filter: string;
}

export interface DatagridContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    advanced: boolean;
    datasource: ListValue;
    refreshInterval: number;
    itemSelection?: SelectionSingleValue | SelectionMultiValue;
    itemSelectionMethod: ItemSelectionMethodEnum;
    itemSelectionMode: ItemSelectionModeEnum;
    showSelectAllToggle: boolean;
<<<<<<< HEAD
    loadingType: LoadingTypeEnum;
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    columns: ColumnsType[];
    columnsFilterable: boolean;
    pageSize: number;
    pagination: PaginationEnum;
<<<<<<< HEAD
    showPagingButtons: ShowPagingButtonsEnum;
    showNumberOfRows: boolean;
    pagingPosition: PagingPositionEnum;
=======
    pagingPosition: PagingPositionEnum;
    showPagingButtons: ShowPagingButtonsEnum;
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    loadMoreButtonCaption?: DynamicValue<string>;
    showEmptyPlaceholder: ShowEmptyPlaceholderEnum;
    emptyPlaceholder?: ReactNode;
    rowClass?: ListExpressionValue<string>;
    onClickTrigger: OnClickTriggerEnum;
    onClick?: ListActionValue;
    onSelectionChange?: ActionValue;
    columnsSortable: boolean;
    columnsResizable: boolean;
    columnsDraggable: boolean;
    columnsHidable: boolean;
    configurationStorageType: ConfigurationStorageTypeEnum;
    configurationAttribute?: EditableValue<string>;
<<<<<<< HEAD
    storeFiltersInPersonalization: boolean;
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    filterList: FilterListType[];
    filtersPlaceholder?: ReactNode;
    filterSectionTitle?: DynamicValue<string>;
    exportDialogLabel?: DynamicValue<string>;
    cancelExportLabel?: DynamicValue<string>;
    selectRowLabel?: DynamicValue<string>;
}

export interface DatagridPreviewProps {
    /**
     * @deprecated Deprecated since version 9.18.0. Please use class property instead.
     */
    className: string;
    class: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
<<<<<<< HEAD
    renderMode?: "design" | "xray" | "structure";
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    advanced: boolean;
    datasource: {} | { caption: string } | { type: string } | null;
    refreshInterval: number | null;
    itemSelection: "None" | "Single" | "Multi";
    itemSelectionMethod: ItemSelectionMethodEnum;
    itemSelectionMode: ItemSelectionModeEnum;
    showSelectAllToggle: boolean;
<<<<<<< HEAD
    loadingType: LoadingTypeEnum;
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    columns: ColumnsPreviewType[];
    columnsFilterable: boolean;
    pageSize: number | null;
    pagination: PaginationEnum;
<<<<<<< HEAD
    showPagingButtons: ShowPagingButtonsEnum;
    showNumberOfRows: boolean;
    pagingPosition: PagingPositionEnum;
=======
    pagingPosition: PagingPositionEnum;
    showPagingButtons: ShowPagingButtonsEnum;
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    loadMoreButtonCaption: string;
    showEmptyPlaceholder: ShowEmptyPlaceholderEnum;
    emptyPlaceholder: { widgetCount: number; renderer: ComponentType<{ children: ReactNode; caption?: string }> };
    rowClass: string;
    onClickTrigger: OnClickTriggerEnum;
    onClick: {} | null;
    onSelectionChange: {} | null;
    columnsSortable: boolean;
    columnsResizable: boolean;
    columnsDraggable: boolean;
    columnsHidable: boolean;
    configurationStorageType: ConfigurationStorageTypeEnum;
    configurationAttribute: string;
<<<<<<< HEAD
    storeFiltersInPersonalization: boolean;
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    onConfigurationChange: {} | null;
    filterList: FilterListPreviewType[];
    filtersPlaceholder: { widgetCount: number; renderer: ComponentType<{ children: ReactNode; caption?: string }> };
    filterSectionTitle: string;
    exportDialogLabel: string;
    cancelExportLabel: string;
    selectRowLabel: string;
}
