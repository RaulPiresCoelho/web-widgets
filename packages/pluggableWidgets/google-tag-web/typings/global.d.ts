type DojoConnectHandle = string & {__dojo__: string};

declare global {
    interface Window {
        dataLayer: any[];
        mx: {
            ui: {
                getContentForm(): { path: string; url: string; title: string };
            };
            session: {
                getSessionObjectId(): string,
<<<<<<< HEAD
                getConfig() : { locale: { code: string } },
                getUserId(): string
=======
                getConfig() : { locale: { code: string } }
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
            }
        };
        dojo: {
            connect: (o: object, f: string, cb: () => void) => DojoConnectHandle;
            disconnect: (DojoConnectHandle) => void;
        }
    }
    interface Document {
        mxGtag?: {
            ensureGtagIncluded: (tagId: string) => boolean;
            getGtag: () => (...args: any[]) => void;
        };
    }
}

export {};
