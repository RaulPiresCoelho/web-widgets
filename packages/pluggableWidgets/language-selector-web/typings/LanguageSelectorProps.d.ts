/**
 * This file was generated from LanguageSelector.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Framework Team
 */
import { CSSProperties } from "react";
<<<<<<< HEAD
import { DynamicValue, ListValue, ListExpressionValue } from "mendix";
=======
import { ListValue, ListExpressionValue } from "mendix";
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)

export type PositionEnum = "left" | "right" | "top" | "bottom";

export type TriggerEnum = "click" | "hover";

export interface LanguageSelectorContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    languageOptions: ListValue;
    languageCaption: ListExpressionValue<string>;
    position: PositionEnum;
    trigger: TriggerEnum;
    hideForSingle: boolean;
<<<<<<< HEAD
    screenReaderLabelCaption?: DynamicValue<string>;
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
}

export interface LanguageSelectorPreviewProps {
    /**
     * @deprecated Deprecated since version 9.18.0. Please use class property instead.
     */
    className: string;
    class: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
<<<<<<< HEAD
    renderMode?: "design" | "xray" | "structure";
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    languageOptions: {} | { caption: string } | { type: string } | null;
    languageCaption: string;
    position: PositionEnum;
    trigger: TriggerEnum;
    hideForSingle: boolean;
<<<<<<< HEAD
    screenReaderLabelCaption: string;
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
}
