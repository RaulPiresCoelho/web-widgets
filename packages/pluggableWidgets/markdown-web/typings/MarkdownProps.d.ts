/**
 * This file was generated from Markdown.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Framework Team
 */
import { EditableValue } from "mendix";

export interface MarkdownContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    stringAttribute: EditableValue<string>;
}

export interface MarkdownPreviewProps {
    readOnly: boolean;
<<<<<<< HEAD
    renderMode?: "design" | "xray" | "structure";
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    stringAttribute: string;
}
