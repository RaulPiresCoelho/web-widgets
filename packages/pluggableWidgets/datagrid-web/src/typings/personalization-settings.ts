<<<<<<< HEAD
import { FilterData } from "@mendix/widget-plugin-filtering/typings/settings";
=======
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
import { ColumnId } from "./GridColumn";
import { SortDirection, SortRule } from "./sorting";

export interface ColumnPersonalizationSettings {
    columnId: ColumnId;
    size: number | undefined;
    hidden: boolean;
    orderWeight: number;
    sortDir: SortDirection | undefined;
    sortWeight: number | undefined;
<<<<<<< HEAD
=======
    filterSettings: unknown;
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
}

interface ColumnPersonalizationStorageSettings {
    columnId: ColumnId;
    size: number | undefined;
    hidden: boolean;
<<<<<<< HEAD
}

export type ColumnFilterSettings = Array<[key: ColumnId, data: FilterData]>;

export type GroupFilterSettings = Array<[key: string, data: FilterData]>;

export interface GridPersonalizationStorageSettings {
    name: string;
    schemaVersion: 2;
    settingsHash: string;
    columns: ColumnPersonalizationStorageSettings[];
    groupFilters: GroupFilterSettings;
    columnFilters: ColumnFilterSettings;
=======
    filterSettings: unknown;
}

export interface GridPersonalizationStorageSettings {
    name: string;
    schemaVersion: number;
    settingsHash: string;

    columns: ColumnPersonalizationStorageSettings[];
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    columnOrder: ColumnId[];
    sortOrder: SortRule[];
}
