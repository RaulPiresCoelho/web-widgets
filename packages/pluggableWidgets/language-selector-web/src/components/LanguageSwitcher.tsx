<<<<<<< HEAD
import { FloatingFocusManager } from "@floating-ui/react";
import classNames from "classnames";
import { createElement, CSSProperties, ReactElement, useState } from "react";
import { PositionEnum, TriggerEnum } from "../../typings/LanguageSelectorProps";
import { useFloatingUI } from "../hooks/useFloatingUI";
import { LanguageItem } from "../LanguageSelector";

export interface LanguageSwitcherProps {
    className: string;
    currentLanguage: LanguageItem | undefined;
    languageList: LanguageItem[];
    onSelect?: (lang: LanguageItem) => void;
    position: PositionEnum;
    screenReaderLabelCaption?: string;
    style?: CSSProperties;
    tabIndex: number;
    trigger: TriggerEnum;
}

export const LanguageSwitcher = ({
    className,
    currentLanguage,
    languageList,
    onSelect,
    position,
    screenReaderLabelCaption,
    style,
    tabIndex,
    trigger
}: LanguageSwitcherProps): ReactElement => {
    const [isOpen, setOpen] = useState(false);

    const {
        activeIndex,
        context,
        floatingStyles,
        getFloatingProps,
        getItemProps,
        getReferenceProps,
        handleSelect,
        isTypingRef,
        listRef,
        refs
    } = useFloatingUI({
        currentLanguage,
        isOpen,
        languageList,
        onSelect,
        position,
        setOpen,
        triggerOn: trigger
    });

    return (
        <div className={classNames(className, "widget-language-selector", "popupmenu")} style={style}>
            <div
                ref={refs?.setReference}
                className={"popupmenu-trigger popupmenu-trigger-alignment"}
                aria-label={screenReaderLabelCaption || "Language selector"}
                aria-autocomplete="none"
                aria-activedescendant={isOpen && activeIndex !== null ? "" : undefined}
                tabIndex={tabIndex}
                {...getReferenceProps?.()}
            >
                <span className="current-language-text">{currentLanguage?.value || ""}</span>
                <span className="language-arrow" aria-hidden="true">
                    <div className={`arrow-image ${isOpen ? "arrow-up" : "arrow-down"}`} />
                </span>
            </div>
            {isOpen && (
                <FloatingFocusManager context={context!} modal={false}>
                    <div
                        aria-activedescendant={isOpen && activeIndex !== null ? "" : undefined}
                        className="popupmenu-menu"
                        ref={refs?.setFloating}
                        style={{
                            ...floatingStyles,
                            outline: 0,
                            overflowY: "auto"
                        }}
                        {...getFloatingProps?.()}
                    >
                        {languageList.map((item, index) => (
                            <div
                                key={item._guid}
                                ref={node => {
                                    listRef.current[index] = node;
                                }}
                                role="option"
                                tabIndex={index === activeIndex ? 0 : -1}
                                className={classNames("popupmenu-basic-item", {
                                    active: currentLanguage === item,
                                    highlighted: activeIndex === index
                                })}
                                {...getItemProps?.({
                                    onKeyDown(event) {
                                        if (event.key === "Enter") {
                                            event.preventDefault();
                                            handleSelect(index);
                                        }

                                        if (event.key === " " && !isTypingRef.current) {
                                            event.preventDefault();
                                            handleSelect(index);
                                        }
                                    },
                                    onClick() {
                                        handleSelect(index);
                                    }
                                })}
                            >
                                {item.value}
                            </div>
                        ))}
                    </div>
                </FloatingFocusManager>
            )}
        </div>
    );
};
=======
import classNames from "classnames";
import { createElement, ReactElement, useEffect, useRef, useState, useCallback, SyntheticEvent } from "react";
import {
    isBehindElement,
    isBehindRandomElement,
    isElementPartiallyOffScreen,
    isElementVisibleByUser,
    moveAbsoluteElementOnScreen,
    unBlockAbsoluteElementBottom,
    unBlockAbsoluteElementLeft,
    unBlockAbsoluteElementRight,
    unBlockAbsoluteElementTop,
    handleOnClickOutsideElement
} from "../utils/document";
import { PositionEnum, TriggerEnum } from "../../typings/LanguageSelectorProps";
import { LanguageItem } from "../LanguageSelector";

export interface LanguageSwitcherProps {
    preview: boolean;
    currentLanguage: LanguageItem | undefined;
    languageList: LanguageItem[];
    position: PositionEnum;
    onSelect?: (lang: LanguageItem) => void;
    trigger: TriggerEnum;
    className: string;
}
export const LanguageSwitcher = (props: LanguageSwitcherProps): ReactElement => {
    const { languageList, trigger } = props;
    const [visibility, setVisibility] = useState(false);
    const ref = useRef<HTMLDivElement>(null);

    if (!props.preview) {
        handleOnClickOutsideElement(ref, () => setVisibility(false));
    }

    const onClickHandle = useCallback(
        (e: SyntheticEvent): void => {
            e.preventDefault();
            e.stopPropagation();
            setVisibility(prev => !prev);
        },
        [setVisibility]
    );

    const onHover =
        trigger === "hover"
            ? {
                  onMouseEnter: onClickHandle,
                  onMouseLeave: onClickHandle
              }
            : {};

    const onClick = trigger === "click" ? { onClick: onClickHandle } : {};

    useEffect(() => {
        const element = ref.current?.querySelector(".popupmenu-menu") as HTMLDivElement | null;
        if (element) {
            element.style.display = visibility ? "flex" : "none";
            if (visibility) {
                correctPosition(element, props.position);
            }
        }
    }, [props.position, visibility]);

    return (
        <div ref={ref} className={classNames(props.className, "widget-language-selector", "popupmenu")} {...onHover}>
            <div className={"popupmenu-trigger popupmenu-trigger-alignment"} {...onClick}>
                <span className="current-language-text">{props.currentLanguage?.value || ""}</span>
                <span className="language-arrow" aria-hidden="true">
                    <div className={`arrow-image ${visibility ? "arrow-up" : "arrow-down"}`} />
                </span>
            </div>
            <div className={classNames("popupmenu-menu", `popupmenu-position-${props.position}`)}>
                {languageList.map(item => (
                    <div
                        key={item._guid}
                        className={`popupmenu-basic-item ${
                            item._guid === props.currentLanguage?._guid ? "active" : ""
                        }`}
                        onClick={() => {
                            if (props.onSelect) {
                                return props.onSelect(item);
                            }
                        }}
                    >
                        {item.value}
                    </div>
                ))}
            </div>
        </div>
    );
};

function correctPosition(element: HTMLElement, position: PositionEnum): void {
    const dynamicDocument: Document = element.ownerDocument;
    const dynamicWindow = dynamicDocument.defaultView as Window;
    let boundingRect: DOMRect = element.getBoundingClientRect();
    const isOffScreen = isElementPartiallyOffScreen(dynamicWindow, boundingRect);
    if (isOffScreen) {
        moveAbsoluteElementOnScreen(dynamicWindow, element, boundingRect);
    }

    boundingRect = element.getBoundingClientRect();
    const blockingElement = isBehindRandomElement(dynamicDocument, element, boundingRect, 3, "popupmenu");
    if (blockingElement && isElementVisibleByUser(dynamicDocument, dynamicWindow, blockingElement)) {
        unBlockAbsoluteElement(element, boundingRect, blockingElement.getBoundingClientRect(), position);
    } else if (blockingElement) {
        let node = blockingElement;
        do {
            if (isBehindElement(element, node, 3) && isElementVisibleByUser(dynamicDocument, dynamicWindow, node)) {
                return unBlockAbsoluteElement(element, boundingRect, node.getBoundingClientRect(), position);
            } else if (node.parentElement) {
                node = node.parentElement as HTMLElement;
            } else {
                break;
            }
        } while (node.parentElement);
    }
}

function unBlockAbsoluteElement(
    element: HTMLElement,
    boundingRect: DOMRect,
    blockingElementRect: DOMRect,
    position: PositionEnum
): void {
    switch (position) {
        case "left":
            unBlockAbsoluteElementLeft(element, boundingRect, blockingElementRect);
            unBlockAbsoluteElementBottom(element, boundingRect, blockingElementRect);
            break;
        case "right":
            unBlockAbsoluteElementRight(element, boundingRect, blockingElementRect);
            unBlockAbsoluteElementBottom(element, boundingRect, blockingElementRect);
            break;
        case "top":
            unBlockAbsoluteElementTop(element, boundingRect, blockingElementRect);
            break;
        case "bottom":
            unBlockAbsoluteElementBottom(element, boundingRect, blockingElementRect);
            break;
    }
}
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
