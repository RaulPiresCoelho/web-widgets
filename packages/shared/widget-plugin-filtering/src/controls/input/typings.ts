<<<<<<< HEAD
import { InputFilterInterface } from "../../typings/InputFilterInterface";
import { InputStore } from "../../stores/InputStore";
import { AllFunctions } from "../../typings/FilterFunctions";
=======
export type Optional<T> = T | undefined;
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)

export interface BaseProps {
    adjustable: boolean;
    className?: string;
    id?: string;
    name: string;
    placeholder?: string;
    screenReaderButtonCaption?: string;
    screenReaderInputCaption?: string;
    styles?: React.CSSProperties;
    tabIndex?: number;
<<<<<<< HEAD
    type: "text" | "number";
    badge?: string;
}

export type FilterFnList<T> = Array<{ value: T; label: string }>;

export interface InputProps<Fn extends AllFunctions> {
    onFilterChange: (filterFn: Fn) => void;
    filterFn: Fn;
    filterFnList: FilterFnList<Fn>;
    inputStores: [InputStore, InputStore];
    disableInputs?: boolean;
    inputRef: React.RefAttributes<HTMLInputElement>["ref"];
    defaultValue?: string;
}

export interface InputComponentProps<Fn extends AllFunctions> extends BaseProps, InputProps<Fn> {}

export interface InputHookProps<Fn, V> {
    filters: FilterFnList<Fn>;
    defaultFilter: Fn;
    defaultValue: V | undefined;
    changeDelay?: number;
    filterStore: InputFilterInterface;
    disableInputs?: (filterFn: Fn) => boolean;
=======
}

export type FilterList<T> = Array<{ value: T; label: string }>;

export interface InputProps<TFilterEnum> {
    onFilterChange: (type: TFilterEnum, isFromUserInteraction: boolean) => void;
    defaultFilter: TFilterEnum;
    filters: FilterList<TFilterEnum>;

    onInputChange: React.ChangeEventHandler<HTMLInputElement>;
    inputType: "text" | "number";
    inputDisabled?: boolean;
    inputRef?: React.RefAttributes<HTMLInputElement>["ref"];
    inputValue: string;
}

export interface InputComponentProps<TFilterEnum> extends BaseProps, InputProps<TFilterEnum> {}

export interface ValueHelper<T> {
    /** Function to map input string to value. */
    fromString(arg: string): T;
    /** Function to map value to string. */
    toString(arg: T): string;
    /** Function to compare the previous value and the new value. */
    equals(a: T, b: T): boolean;
}

export interface InputHookProps<TValue, TFilterEnum> {
    defaultFilter: TFilterEnum;
    filters: FilterList<TFilterEnum>;
    value: Optional<TValue>;
    onChange: (value: Optional<TValue>, filter: TFilterEnum) => void;
    valueHelper: ValueHelper<TValue>;
    changeDelay?: number;
    parentChannelName: string | null;
    name: string;
    inputDisabled?: (filter: TFilterEnum) => boolean;
    inputType: "text" | "number";
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
}
