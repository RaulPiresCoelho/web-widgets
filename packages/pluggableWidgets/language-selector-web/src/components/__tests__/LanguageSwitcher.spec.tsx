<<<<<<< HEAD
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { createElement } from "react";
import { PositionEnum, TriggerEnum } from "typings/LanguageSelectorProps";
import { LanguageSwitcher, LanguageSwitcherProps } from "../LanguageSwitcher";
import "@testing-library/jest-dom";

jest.useFakeTimers();

let props: LanguageSwitcherProps = {
=======
import { render } from "@testing-library/react";
import { createElement } from "react";
import { PositionEnum, TriggerEnum } from "typings/LanguageSelectorProps";
import { LanguageSwitcher, LanguageSwitcherProps } from "../LanguageSwitcher";

let props: LanguageSwitcherProps = {
    preview: false,
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    currentLanguage: undefined,
    languageList: [],
    position: "left" as PositionEnum,
    onSelect: jest.fn(),
    trigger: "click" as TriggerEnum,
<<<<<<< HEAD
    className: "",
    tabIndex: 0
=======
    className: ""
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
};
const language = { _guid: "111", value: "En us" };

describe("Language switcher", () => {
<<<<<<< HEAD
    it("renders the structure with empty language list", async () => {
        const { asFragment } = render(<LanguageSwitcher {...props} />);
        const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
        const triggerElement = screen.getByRole("combobox");

        await user.click(triggerElement);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders the structure with language list and selected default language", async () => {
        props = { ...props, languageList: [language], currentLanguage: language };
        const { asFragment } = render(<LanguageSwitcher {...props} />);
        const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
        const triggerElement = screen.getByRole("combobox");

        await user.click(triggerElement);
=======
    it("renders the structure with empty language list", () => {
        const { asFragment } = render(<LanguageSwitcher {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders the structure with language list and selected default language", () => {
        props = { ...props, languageList: [language], currentLanguage: language };
        const { asFragment } = render(<LanguageSwitcher {...props} />);
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
        expect(asFragment()).toMatchSnapshot();
    });
});
