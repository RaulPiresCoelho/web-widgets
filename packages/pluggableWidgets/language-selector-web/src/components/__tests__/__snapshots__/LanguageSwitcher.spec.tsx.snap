// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language switcher renders the structure with empty language list 1`] = `
<DocumentFragment>
  <div
    class="widget-language-selector popupmenu"
  >
    <div
<<<<<<< HEAD
      aria-activedescendant=""
      aria-autocomplete="none"
      aria-controls=":r0:"
      aria-expanded="true"
      aria-haspopup="listbox"
      aria-label="Language selector"
      class="popupmenu-trigger popupmenu-trigger-alignment"
      data-floating-ui-inert=""
      role="combobox"
      tabindex="0"
=======
      class="popupmenu-trigger popupmenu-trigger-alignment"
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    >
      <span
        class="current-language-text"
      />
      <span
        aria-hidden="true"
        class="language-arrow"
      >
        <div
<<<<<<< HEAD
          class="arrow-image arrow-up"
=======
          class="arrow-image arrow-down"
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
        />
      </span>
    </div>
    <div
<<<<<<< HEAD
      aria-activedescendant=""
      aria-orientation="vertical"
      class="popupmenu-menu"
      data-floating-ui-focusable=""
      id=":r0:"
      role="listbox"
      style="position: fixed; left: 0px; top: 0px; outline: 0; overflow-y: auto; transform: translate(0px, 0px);"
      tabindex="0"
=======
      class="popupmenu-menu popupmenu-position-left"
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    />
  </div>
</DocumentFragment>
`;

exports[`Language switcher renders the structure with language list and selected default language 1`] = `
<DocumentFragment>
  <div
    class="widget-language-selector popupmenu"
  >
    <div
<<<<<<< HEAD
      aria-activedescendant=""
      aria-autocomplete="none"
      aria-controls=":r2:"
      aria-expanded="true"
      aria-haspopup="listbox"
      aria-label="Language selector"
      class="popupmenu-trigger popupmenu-trigger-alignment"
      data-floating-ui-inert=""
      role="combobox"
      tabindex="0"
=======
      class="popupmenu-trigger popupmenu-trigger-alignment"
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
    >
      <span
        class="current-language-text"
      >
        En us
      </span>
      <span
        aria-hidden="true"
        class="language-arrow"
      >
        <div
<<<<<<< HEAD
          class="arrow-image arrow-up"
=======
          class="arrow-image arrow-down"
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
        />
      </span>
    </div>
    <div
<<<<<<< HEAD
      aria-activedescendant=""
      aria-orientation="vertical"
      class="popupmenu-menu"
      data-floating-ui-focusable=""
      id=":r2:"
      role="listbox"
      style="position: fixed; left: 0px; top: 0px; outline: 0; overflow-y: auto; transform: translate(0px, 0px);"
      tabindex="-1"
    >
      <div
        class="popupmenu-basic-item active highlighted"
        role="option"
        tabindex="0"
=======
      class="popupmenu-menu popupmenu-position-left"
    >
      <div
        class="popupmenu-basic-item active"
>>>>>>> daa3fce04 (Add DE localization to rich-text-web)
      >
        En us
      </div>
    </div>
  </div>
</DocumentFragment>
`;
